
<!DOCTYPE html>
<html>
  <head>
    <title><b>Building Pipeline for Bioinformatics Data using PluMA</b></title>

  </head>
  <body>

<h1><b>OVERVIEW AND BACKGROUND</b></h1>
      <p>The objective of PluMA is to make it easy for developers to create adaptable and lightweight analysis pipelines. These pipelines allow developers to incorporate new algorithms in their preferred programming language and debug them alongside other stages in the pipeline that might use different file formats and programming languages. To achieve this goal, PluMA uses plugins which are available in a large collection in its plugin pool. These plugins are written in various programming languages and can be executed sequentially to form a pipeline. PluMA allows users to easily add, remove, or replace plugins using its user interface. Since plugins are dynamically loaded, users can download PluMA and only the plugins they require from the plugin pool to assemble their pipeline. Furthermore, PluMA provides plugins for both CPU and GPU implementations.</p>
<hr>

<h1><b>GOAL</b></h1>
       <p>Construct a Pipeline for the Paper </p>
<a href="https://www.nature.com/articles/s41586-020-2095-1">Microbiome analyses of blood and tissues suggest cancer
diagnostic approach | Nature</a>
<hr>
       
 <h1><b> OBJECTIVE <b></h1>
        <p> The objective of PluMA is to simplify the creation of adaptable and lightweight analysis pipelines that can integrate new algorithms in different programming languages and file formats. PluMA uses plugins, with a collection of pre-existing plugins available, implemented in various programming languages and for both CPU and GPU, that can be run sequentially to form a pipeline. The plugins can be easily added, removed, or replaced through a user interface, and dynamically loaded, enabling users to assemble pipelines quickly and efficiently. Transforming microbiome analysis from individual to community-driven. </p>
<hr>
        
<h1><b> MOTIVATION </b></h1>
        <p> The Motivation of PluMA is: </p>
          <pre>
          Interoperability of programming languages.
          Handling different file formats.
          Testing and debugging within a larger pipeline.
          Flexibility and lightweightness.
          </pre>
<hr>
        
<h1><b>CHALLENGES</b></h1>
         <p>Few Challenges of PluMA is:</p>
          <pre>
          Clone the Repository.
          Create and build required config File.
          Test the Built Plugins using single input.
          Run them with Multiple inputs using kitty.
         </pre>
<hr>
	
<h1><b>DATA SOURCES</b></h1>
         <p> Paper worked on :</p>
<a href="https://www.nature.com/articles/s41586-020-2095-1">Microbiome analyses of blood and tissues suggest cancer
diagnostic approach | Nature</a>
          <pre>
          Authors: “Gregory D. Poore, Evguenia Kopylova, Qiyun Zhu, Carolina Carpenter, Serena Fraraccio, Stephen Wandro, Tomasz Kosciolek, Stefan Janssen, Jessica Metcalf, Se Jin Song, Jad Kanbar, Sandrine Miller-Montgomery, Robert Heaton, Rana Mckay, Sandip Pravin Patel, Austin D. Swafford & Rob Knight”
         </pre>
<hr>
<h1><b>DESCRIPTION</b></h1>
      <p> The researchers discovered that they could identify specific patterns of microorganisms in both cancer tissues and blood samples from cancer patients. These patterns, called microbial signatures, were still able to predict the presence of cancer even in early-stage cases and in situations where traditional genetic tests may not be effective. In other words, these microbial signatures were found in both cancer tissues and blood samples, and they were still able to accurately indicate the presence of cancer even in early stages of the disease or when other standard genetic tests may not provide reliable results. Considering the 5 samples each from the below mentioned data from European Nucleotide Archive.
</p>
<pre>
Human Non-Reference Catalog- Plasma microbiome (HIV-Free)
Plasma microbiome in prostate cancer (PC)
Skin cutaneous melanoma (SKCM) and lung cancer (LC)
</pre>
	  <hr>
      <h1><b>METHODS AND TOOLS</b></h1>
      <p>Below Mentioned are the list of Plugins and why they are used to build pipeline for the Paper.</p>
      <style>
      <table>
{
  font-family: arial, sans-serif;
  border-collapse: collapse;
  width: 100%;
}

td, th {
  border: 1px solid #dddddd;
  text-align: left;
  padding: 8px;
}

tr:nth-child(even) {
  background-color: #dddddd;
}
</style>
          <tr>
            <th>Plugins</th>
            <th>What it does?</th>
          </tr>
     
          <tr>
            <td>Kraken2</td>
 	    <td>Map sequences to taxa and counts</td>
</tr>
<tr>
<td>Kraken2Translate</td>
 <td>translates .krak2 file from .kraken2</td>
</tr>
<tr>
  <td>Kraken2Krona</td>
 <td>visualize taxonomic or hierarchical data</td>
</tr>
<tr>
<td>KronaTree</td>
   <td>Biological Classification of Data</td>
</tr>
<tr>
<td>KronaCSV</td>
  <td>Converts Krona files to CSV</td>
</tr>
<tr>
<td>CSVMerge</td>
 <td>Merges the CSV files</td>
</tr>
<tr>
 <td>KronaGram</td>
  <td>Visualization</td>
</tr>
      </table>
<hr>

<h1>RESULTS</h1>
      <p>
Kraken2: Input for Kraken2 plugin is a textfile which contains the DNA Sequencing in fastq file format and the output is in [.krak2] format. Kraken to Maps the sequences to human or non-human taxa. 
</p>
<pre>Input Plugin format is written as below.</pre>
<pre>Output is shown as below the sequences are classified into human [87.13%] and bacterial [12.87%]</pre>
<p>
Kraken2Translate: Input for Kraken2Translate plugin is a parameters.kraken2translate.txt which contains the output of Kraken2 with .Krak2 file format and the output is in [.kraken2] format. Kraken2Translate is a file converter to convert .Krak2 file to .Kraken2
</p>
<pre>Input Plugin format and text file format is written as below.
</pre>
<pre>Output is a translated .Kraken2 file format file of input .krak2 in txt format
</pre>
<hr>

<h1><b>CONCLUSION</b></h1>
      <p>Below mentioned is a pipeline for the Paper “Microbiome analyses of blood and tissues suggest cancer diagnostic approach | Nature” when raw DNA sequences are given as input and this gives us the classified samples for each biological classification and the types of microbial signatures that come under each classification to easily identify which sample has which microbial signature present in his blood and tissues.So, when we considering healthy DNA samples and if any of the sample is classified to these microbial signatures that means there is a probability that the person is affected with cancer and is in the initial stage Ia or IIC and can be diagnosed in the early stage.</p>
<hr>

<h1>FUTURE WORK</h1>
      <p>As we don’t have the data to classify the microbial signature into types of cancer future work is required to identify what type of cancer is present in the DNA Sequences. Below models can be used in future work.</p>
<pre>
Lefse (for a better idea of differentiating taxa)
Networks (for ecological relationships)
Machine Learning (with taxa abundances as features)
</pre>
<hr>

<h1><b>CONTRIBUTIONS</b></h1>
      <pre>
Prof Trevor Cickovski My mentor, I thank him for supporting and helping me to understand the paper and the biological terminology and helped me understand how PluMA works and how it should be used. 
 
Dr. Giri Narasimhan Our Project head, I thank him for helping me in improving my Presentation skills and how to present my work and what to present. 
</pre>
<hr>

<h1><b>REFERENCES</b></h1>
<p><a href="https://biorg.cis.fiu.edu/pluma/plugins.html ">PluMA Plugin pool</a></p>
<p><<a href="https://www.nature.com/articles/s41586-020-2095-1">Microbiome analyses of blood and tissues suggest cancer
diagnostic approach | Nature</a></p>

</body>
</html>
  
